### DISCUSSION TITLE
How would I implement an `extra QEMU options` flag in Nu?

### DISCUSSION THREAD

**Discussion by J-Cake** [nushell/nushell#12824] [Q&A] [answered]
So in order to allow me to pass extra arguments to Qemu, I've tried the following:

```nu
def main [--drive: path, ...extra: string] {
    qemu-system-x86_64 
        --drive $drive # not a real implemenation
        ...$extra 
}
```

This is problematic because Nu recognises the the `--cdrom` option I'd like to pass it as a flag to be interpreted by `main`, and because I never defined an option called that, it errors.

Next I tried a list of strings:

```nu
def main [--drive: path, extra: list] {
    # same as above
}

main --drive ./cheese.qcow2 ['--cdrom' 'os.raw']
```

which failed for the same reason. 

What might be the best way to pass an argument-looking value to a function without it being interpreted as such?

**Comment by devyn** [comment, marked as answer]
Hi, you're looking for `def --wrapped`. This option allows rest args to absorb unknown flags as well, and it's meant exactly for this purpose.

**Reply by J-Cake** [reply]
Thanksss I never thought to look there :D
